subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -174,125 +247,132 @@ static int pd_decode_command(struct osdp_pd *pd, uint8_t *buf, int len)
 		ret = OSDP_PD_ERR_NONE;
 		break;
 	case CMD_OUT:
-		if (len != CMD_OUT_DATA_LEN) {
+		if (len != CMD_OUT_DATA_LEN || !pd->command_callback) {
 			break;
 		}
-		cmd = osdp_cmd_alloc(pd);
-		if (cmd == NULL) {
-			LOG_ERR("cmd alloc error");
+		cmd.id = OSDP_CMD_OUTPUT;
+		cmd.output.output_no = buf[pos++];
+		cmd.output.control_code = buf[pos++];
+		cmd.output.timer_count = buf[pos++];
+		cmd.output.timer_count |= buf[pos++] << 8;
+		ret = pd->command_callback(pd->command_callback_arg, &cmd);
+		if (ret != 0) {
```

Comment:
This `if `is copy&pasted several times.
Could you please consider small refactoring.

Reply:
Sure.